tests/x11/yast2_snapper.pm

```diff
@@ -0,0 +1,92 @@
+use base "x11test";
+use testapi;
+
+# Test for basic yast2-snapper functionality. It assumes the data of the
+# opensuse distri to be available at /home/$username/data (as granted by
+# console_setup.pm)
+
+# Helper for letting y2-snapper to create a snapper snapshot
+sub y2snapper_create_snapshot() {
+    my $self = shift;
+    my $name = shift || "Awesome Snapshot";
+   # Open the 'C'reate dialog and wait until it is there
+    send_key "alt-c";
+    assert_screen 'yast2_snapper-createsnapshotdialog', 100;
+    # Fill the form and finish by pressing the 'O'k-button
+    type_string $name;
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    send_key "tab";
+    sleep 2;
+    type_string "a=1,b=2";
+    sleep 2;
+    send_key "alt-o";
+}
+sub run() {
+    my $self = shift;
+    # Make sure yast2-snapper is installed (if not: install it)
+    ensure_installed "yast2-snapper";
+    # Start an xterm as root
+    x11_start_program("xterm");
+    wait_idle;
+    become_root;
+    # Create a snapper config
+    script_run "btrfs subvolume create /tmp/testdata";
+    script_run "snapper create-config /tmp/testdata";
+    # Start the yast2 snapper module and wait until it is started
+    script_run "yast2 snapper";
+    assert_screen 'yast2_snapper-nosnapshots', 100;
+    # Create a new snapshot
+    $self->y2snapper_create_snapshot();
+    # Make sure the snapshot is listed in the main window
+    assert_screen 'yast2_snapper-snapshotlisted', 100;
+    # C'l'ose  the snapper module
+    send_key "alt-l";
+    # Download & untar test files
+    wait_idle;
+    script_run "cd /tmp";
+    script_run "tar -xzf /home/$username/data/yast2_snapper.tgz";
+    wait_idle;
```

Comment:
This will probably work fine, but just as a learning exercise, I'd do this as 

script_run "tar -xzf.... && echo 'tar_complete'";
wait_serial ('tar_complete') || die 'tar -xzf failed'

instead of the wait_idle, just so you *know* the tar has finished properly.